### ISSUE TITLE
Wrong diagnostic for zero-sized array in transparent struct

### ISSUE THREAD

**Issue by lukas-code** [rust-lang/rust#98064]
Given the following code: 

```rust
#[repr(transparent)]
pub struct Aligned<T, A> {
    align: [A; 0],
    value: T,
}
```

The current output is:

```
error[E0690]: transparent struct needs at most one non-zero-sized field, but has 2
 --> kernel/src/align.rs:2:1
  |
2 | pub struct Aligned<T, A> {
  | ^^^^^^^^^^^^^^^^^^^^^^^^ needs at most one non-zero-sized field, but has 2
3 |     align: [A; 0],
  |     ------------- this field is non-zero-sized
4 |     value: T,
  |     -------- this field is non-zero-sized

For more information about this error, try `rustc --explain E0690`.
```

Ideally the output should look like:

```
error[E0691]: zero-sized field in transparent struct has alignment larger than 1
 --> kernel/src/align.rs:3:5
  |
3 |     align: [A; 0],
  |     ^^^^^^^^^^^^^ may have alignment larger than 1
  |
help: consider changing `#[repr(transparent)]` to `#[repr(C)]`
  |
1 - #[repr(transparent)]
1 + #[repr(C)]
  | 

For more information about this error, try `rustc --explain E0691`.
```

**Comment by lukas-code** [comment]
Closing this as fixed, the compiler output now mentions alignment:
```
error[E0690]: transparent struct needs at most one field with non-trivial size or alignment, but has 2
 --> src/lib.rs:2:1
  |
2 | pub struct Aligned<T, A> {
  | ^^^^^^^^^^^^^^^^^^^^^^^^ needs at most one field with non-trivial size or alignment, but has 2
3 |     align: [A; 0],
  |     ------------- this field has non-zero size or requires alignment
4 |     value: T,
  |     -------- this field has non-zero size or requires alignment

For more information about this error, try `rustc --explain E0690`.
error: could not compile `playground` (lib) due to 1 previous error
```
